### PR TITLE
Replace camel- and snake-case keys libs with custom code

### LINKED ISSUE #170: Code fails to minify due to non-transpiled camelcase-keys dependency
The `camelcase-keys` and `snakecase-keys` dependencies [are not designed to transpile](https://github.com/sindresorhus/camelcase-keys/issues/17) into browser-runnable code, so attempting to minify the front end build will result in an error.

Browserify appears to offer the ability to [transpile a library dependency](https://github.com/babel/babel/issues/1110), but it may be simpler just to replace the dependencies with our own key mapping code or eventually remove the need for this at all.

### PR DESCRIPTION
**Pull request by landonreed**
fixes #170.

### CODE REVIEW

File: lib/common/util/map-keys.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// @flow
+import forEach from 'lodash.foreach'
```
Comment (trevorgerhardt) [inline lib/common/util/map-keys.js]: I'd recommend adding the entire lodash as a dependency. Watch out when importing directly though, camelCase counts when you're importing files. 
```js
import forEach from 'lodash.foreach'
/// becomes
import forEach from 'lodash/forEach'

File: lib/common/util/map-keys.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import camelCase from 'lodash.camelcase'
+import isPlainObject from 'lodash.isplainobject'
+import snakeCase from 'lodash.snakecase'
+
+/**
+ * Converts the keys for an object (or array of objects) using string mapping
+ * function passed in. Operates on object recursively.
+ */
+function mapObjectKeys (object: any, keyMapper: (string) => string): any {
```
Comment (trevorgerhardt) [inline lib/common/util/map-keys.js]: I'd write it like this: 

```js
function mapObjectKeys (object, keyMapper) {
  if (Array.isArray(object)) return object.map(v => mapObjectKeys(v, keyMapper))
  return _.mapValues(_.mapKeys(object, keyMapper), (v) => 
    _.isPlainObject(v) || Array.isArray(v) ? mapObjectKeys(v) : v)
}
```

Comment (trevorgerhardt) [reply lib/common/util/map-keys.js]: I think that works :)

Comment (landonreed) [reply lib/common/util/map-keys.js]: Hmm, doesn't work, @trevorgerhardt.

**Review by trevorgerhardt — APPROVED** [review]

**Pull request merged**